### ISSUE TITLE
The results seems not good.

### ISSUE THREAD

**Issue by dongdong092** [neka-nat/image_completion_keras#5]
The result seems not good as the result in the paper.It there just need more training or other tricks?

**Comment by generallc** [comment]
Hello, where did you training the module, CPU or GPU?  I can train it on CPU, but maybe my 2  8G Kingston RAMs can't  offer enough storage capacity，only can train the module in Tc+Td epochs. So the result seems not good  because my computer did not complete the training of the entire model.

**Comment by dongdong092** [comment]
Ooh,I just run the master branch code on my  gpu.I ignored the tc and the td stage.I would try to run the animi branch today.

**Comment by zhangqingbo1** [comment]
Yeah, the master branch code does not involve the tc and the td stage. I am looking forward to your new result on the animi branch.
Could you please tell me the version of your GPU,  CUDA,  Tensorflow-gpu  and  Keras , My computer is nvidia GTX1080,  CUDA8.0  Tensorflow-gpu(1.6.0)  and  Keras (2.1.5) ,but it doesn't  work.

**Comment by generallc** [comment]
Yeah, the master branch code does not involve the tc and the td stage. I am looking forward to your new result on the animi branch.
Could you please tell me the version of your GPU, CUDA, Tensorflow-gpu and Keras , My computer is nvidia GTX1080, CUDA8.0 Tensorflow-gpu(1.6.0) and Keras (2.1.5) ,but it doesn't work for the anim and separate_c_d branch.

**Comment by dongdong092** [comment]
I use M40,cuda8.0,Tensorflow-gpu（1.6.0）and Keras（2.0.8）and it run 100 epochs about 2.5 hours.

**Comment by generallc** [comment]
Oh, thanks, maybe my GPU is only 8G, I‘ll try to change the code for solving the problems.

**Comment by zhangqingbo1** [comment]
@dongdong092   when you finished the training, how do you load the trained weight to predict the test sets. Please give me some advice, thanks!

**Comment by dongdong092** [comment]
I run the other branch .In the 20th opoches,it stops by the following error:
print("%d [D loss: %e] [G mse: %e]" % (n, d_loss, g_loss))
TypeError: must be real number, not list

**Comment by zhangqingbo1** [comment]
I also met this error，because the g_loss is not a real number, I print g_loss separately by the following code, which show g_loss is a list such as 'G mse: [0.0028160308, 0.0028160308, 1.1920933e-07]':
            print('%d G mse:' % n),
            print(g_loss),
            print("[D loss: %e]" %d_loss)

**Comment by zhangqingbo1** [comment]
And could you please tell me your email address?

**Comment by neka-nat** [comment]
Hi.
I will check the "TypeError" and fix it.
Thanks.

**Comment by neka-nat** [comment]
I fixed the "TypeError" and merged to master branch.

**Comment by xieenze** [comment]
I train this model on Street View dataset,my result doesn't good as well. Do anybody achieve good result?

**Comment by Alex-toto** [comment]
@xieenze  I train the model on place2 dataset, the result looks good.  But when I  test images with arbitrary mask shape, the results are too bad. Do you test any images  and how to do it ?

**Comment by micklexqg** [comment]
@Alex-toto , which branch did you use? the master or animi?

**Comment by varoudis** [comment]
@dongdong092 im running very slow compared to your 100Ep @ 2.5Hours.
I use batch size 16 on a TitanV GPU (and dual xeon cpus to spare) and Im getting around 20mins per Epoch... Im also only using the 35000 validation images from places...
